hrms/hr/doctype/job_opening/job_opening.py

```diff
@@ -73,38 +113,17 @@ def update_job_requisition_status(self):
 			job_requisition.save()
 
 	def get_context(self, context):
+		job_applicants = frappe.db.count("Job Applicant", {"job_title": self.name})
+		context.no_of_applications = job_applicants
 		context.parents = [{"route": "jobs", "title": _("All Jobs")}]
+		context.posted_on = pretty_date(self.posted_on)
 
 
-def get_list_context(context):
-	context.title = _("Jobs")
-	context.introduction = _("Current Job Openings")
-	context.get_list = get_job_openings
-
-
-def get_job_openings(
-	doctype, txt=None, filters=None, limit_start=0, limit_page_length=20, order_by=None
-):
-	fields = [
-		"name",
-		"status",
-		"job_title",
-		"description",
-		"publish_salary_range",
-		"lower_range",
-		"upper_range",
-		"currency",
-		"job_application_route",
-	]
-
-	filters = filters or {}
-	filters.update({"status": "Open"})
-
-	if txt:
-		filters.update(
-			{"job_title": ["like", "%{0}%".format(txt)], "description": ["like", "%{0}%".format(txt)]}
-		)
-
-	return frappe.get_all(
-		doctype, filters, fields, start=limit_start, page_length=limit_page_length, order_by=order_by
-	)
+def close_expired_job_openings():
+	today = getdate()
+	for d in frappe.get_all(
+		"Job Opening",
+		filters={"Status": "Open", "closes_on": ["<=", today]},
```

Comment:
This will also pick up openings that don't have a closing date set. Ignore those